Modules/+Sources/Msquared.m

```diff
@@ -0,0 +1,481 @@
+            end
+            obj.updatingVal = false;
+                
+            if failcount > 10
+                error('Failed to set the SolsTiS wavelength ten times, aborting tuning attempt.')
+            end
+                
+            if      ~obj.do_etalon_lock
+                obj.com('lock_wavelength', 'solstis', obj.lockList{1});
+                obj.com('set_etalon_lock', 'solstis', obj.lockList{1});
+            elseif  ~obj.do_wavelength_lock
+                obj.com('lock_wavelength', 'solstis', obj.lockList{1});
+            end
+            
+            obj.getFrequency();
+            
+            if strcmp(module, obj.moduleVIS) && abs(obj.VIS_wavelength - target) > .001 % If we're off with the EMM, we probably didn't initially have a good reading on the diff_wavelength.
+                obj.tune(target)    % Try tuning again.
+            end
+        end
+    end
+    
+    methods     % Pref set methods.
+        function val = set_source_on(obj, val, ~)   % 'Fast' modulation method -- usually the PulseBlaster.
+            assert(~isempty(obj.PulseBlaster),'No IP set!')
+            obj.PulseBlaster.lines(obj.PB_line).state = val;
+        end
+        function val = set_armed(obj, val, ~)       % Checks if the laser is outputting power (i.e. has been armed) and complains if result was contradictory.
+            obj.getFrequency(); 
+            
+            if val ~= (obj.output_monitor > .01)    % If we are setting armed to an incorrect value...
+                val = ~val;
+                if val
+                    warndlg(['Request to arm laser. Laser is off, as output_monitor reads ' num2str(obj.output_monitor) '; please turn the laser on'], 'Arm (Sources.Msquared)')
+                else
+                    warndlg(['Request to blackout laser. Laser is on, as output_monitor reads ' num2str(obj.output_monitor) '; please turn the laser off'], 'Blackout (Sources.Msquared)')
+                end
+            end
+        end
+        
+        function host= set_hwserver_host(obj, host, ~)
+            try
+                obj.hwserver = hwserver(host); %#ok<CPROPLC>
+                
+                % Update laser list. (This piece of code was breaking due to update_settings trying to update hwserver_host, which is in use. Instead hardcode moduleName options.)
+%                 opts = obj.hwserver.get_modules('msquared.');
+%                 mp = obj.get_meta_pref('moduleName');
+%                 if ~isequal(mp.choices,opts) % Only update if different
+%                     if ~isempty(mp.value)
+%                         % If we need to reset it, we also need to re-grab the metapref
+%                         obj.moduleName = '';
+%                         mp = obj.get_meta_pref('moduleName');
+%                     end
+%                     mp.choices = opts;
+%                     obj.set_meta_pref('moduleName', mp);
+%                     notify(obj,'update_settings');
+%                 end
```

Comment:
Do we want to leave this commented-out code in here for a reason?

Reply:
Yes. This code is for pulling available lasers from hwserver, but it was breaking, and I didn't feel like fixing it. It's there in case someone wants to fix it.